lib/common/socket.c

```diff
@@ -1455,6 +1649,13 @@ static void proceed_handshake_picotls(h2o_socket_t *sock)
     case 0:
         next_cb = on_handshake_complete;
         break;
+    case PTLS_ERROR_ASYNC_OPERATION:
+#ifdef PTLS_OPENSSL_HAVE_ASYNC
+        do_ssl_async(sock);
+#else
+        goto Retry;
```

Comment:
How about calling `h2o_abort` instead?

My hope is that we would not need to support the case of `ptls_handshake` returning `PTLS_ERROR_ASYNC_OPERATION` when the openssl backend lacks async support. Or do we need to support that?